lambdas/services/process_mns_message_service.py

```diff
@@ -0,0 +1,140 @@
+                logger.info("Handling GP change notification.")
+                self.handle_gp_change_notification(message)
+
+            elif message.type == MNSNotificationTypes.DEATH_NOTIFICATION.value:
+                logger.info("Handling death status notification.")
+                self.handle_death_notification(message)
+
+        except PdsErrorException:
+            self.send_message_back_to_queue(message)
+
+        except Exception as e:
+            logger.info(
+                f"Unable to process message: {message.id}, of type: {message.type}"
+            )
+            logger.info(f"{e}")
+            return
+
+    def handle_gp_change_notification(self, message: MNSSQSMessage):
+        patient_document_references = self.get_patient_documents(
+            message.subject.nhs_number
+        )
+
+        if len(patient_document_references) < 1:
+            logger.info("Patient is not held in the National Document Repository.")
+            logger.info("Moving on to the next message.")
+            return
+
+        updated_ods_code = self.get_updated_gp_ods(message.subject.nhs_number)
+
+        for reference in patient_document_references:
+            if reference["CurrentGpOds"] is not updated_ods_code:
+                self.dynamo_service.update_item(
+                    table_name=self.table,
+                    key=reference["ID"],
+                    updated_fields={"CurrentGpOds": updated_ods_code},
+                )
+
+        logger.info("Update complete for change of GP")
+
+    def handle_death_notification(self, message: MNSSQSMessage):
+        if self.is_informal_death_notification(message):
+            logger.info(
+                "Patient is deceased - INFORMAL, moving on to the next message."
+            )
+            return
+
+        try:
+            patient_documents = self.get_patient_documents(message.subject.nhs_number)
+
+            if len(patient_documents) < 1:
+                logger.info("Patient is not held in the National Document Repository.")
+                logger.info("Moving on to the next message.")
+                return
+
+            if (
+                message.data["deathNotificationStatus"]
+                == DeathNotificationStatus.FORMAL.value
+            ):
+                self.update_patient_details(
+                    patient_documents, PatientOdsInactiveStatus.DECEASED.value
+                )
+                logger.info("Update complete, patient marked DECE.")
+                return
+
+            if (
+                message.data["deathNotificationStatus"]
+                == DeathNotificationStatus.REMOVED.value
+            ):
+                update_ods_code = self.get_updated_gp_ods(message.subject.nhs_number)
+                self.update_patient_details(patient_documents, update_ods_code)
+                logger.info("Update complete for death notification change.")
+        except ClientError as e:
+            logger.error(f"{e}")
```

Comment:
Any reason we're catching `ClientError` here and not in `handle_gp_change_notification`? Maybe best to move this into `handle_mns_notification`?